shared/src/test/scala/mlscript/DiffTests.scala

```diff
@@ -382,17 +434,22 @@ class DiffTests extends org.scalatest.funsuite.AnyFunSuite with org.scalatest.Pa
                       prefixLength = 3
                     }
                 }
-                showResult(prefixLength)
+                showFirstResult(prefixLength)
             }
 
             // If code generation fails, show the error message.
             results match {
-              case L((message, crashed)) =>
-                if (crashed)
-                  output("Code generation crashed:")
-                else
-                  output("Code generation met an error:")
-                output(s"  $message")
+              case IllFormedCode(message) =>
+                totalRuntimeErrors += 1
+                output("Code generation met an error:")
+                output(s"  ${message}")
+              case Unimplemented(message) =>
+                output("Unable to execute the code:")
+                output(s"  ${message}")
+              case UnexpectedCrash(message) =>
```

Comment:
This message should include the name of the error. For example try making it crash with a `MatchError` — it won't even say it's a match error.

Reply:
Fixed in 22ab793.